execution_engine2.spec

```diff
@@ -67,72 +67,58 @@
-                module name from spec-file followed by '.' and name of funcdef
-                from spec-file corresponding to running method (e.g.
-                'KBaseTrees.construct_species_tree' from trees service);
-            job_id - job id if method is asynchronous (optional field).
-        */
-        typedef structure {
-            timestamp time;
-            string method;
-            job_id job_id;
-        } MethodCall;
-
-        /*
-            call_stack - upstream calls details including nested service calls and
-                parent jobs where calls are listed in order from outer to inner.
+        /* Narrative metadata for a job. All fields are optional.
+            run_id - the Narrative-assigned ID of the job run. 1:1 with a job ID.
+            token_id - the ID of the token used to run the method.
+            tag - the release tag, e.g. dev/beta/release.
+            cell_id - the ID of the narrative cell from which the job was run.
         */
         typedef structure {
-            list<MethodCall> call_stack;
             string run_id;
-        } RpcContext;
+            string token_id;
+            string tag;
+            string cell_id;
+        } Meta;
 
         /*
-            method - service defined in standard JSON RPC way, typically it's
-                module name from spec-file followed by '.' and name of funcdef
-                from spec-file corresponding to running method (e.g.
-                'KBaseTrees.construct_species_tree' from trees service);
-            params - the parameters of the method that performed this call;
+            method - the SDK method to run in module.method format, e.g.
+                'KBaseTrees.construct_species_tree'
+            app_id - the id of the Narrative application (UI) running this job (e.g.
+                repo/name)
+            params - the parameters to pass to the method.
 
             Optional parameters:
             service_ver - specific version of deployed service, last version is
                 used if this parameter is not defined
-            rpc_context - context of current method call including nested call
-                history
-            remote_url - run remote service call instead of local command line
-                execution.
             source_ws_objects - denotes the workspace objects that will serve as a
                 source of data when running the SDK method. These references will
-                be added to the autogenerated provenance.
-            app_id - the id of the Narrative application (UI) running this job (e.g.
-                repo/name)
-            mapping<string, string> meta - user defined metadata to associate with
-                the job.
+                be added to the autogenerated provenance. Must be in UPA format (e.g.
+                6/90/4).
+            meta - Narrative metadata to associate with the job.
```

Comment:
The UJS meta field was arbitrary user supplied data, but ee2 ignored all but 5 keys